src/main/java/org/folio/circulation/domain/OpeningDay.java

```diff
@@ -92,11 +113,16 @@ private JsonArray openingHourToJsonArray() {
   }
 
   public JsonObject toJson() {
```

Comment:
Why does an opening day need to be converted to JSON?

Reply:
To be returned as part of `OpeningPeriod` in a mocked response from `/calendar/periods` in tests.

Reply:
I don't think the production domain code should include code used for constructing mock responses (which I would prefer worked differently in general)

Reply:
> which I would prefer worked differently in general

What do you mean?

Reply:
The API tests in mod-circulation were designed to be run either using a set of fake modules or using real modules.

This was intended to reveal integration issues that fakes might obscure e.g. validation, queries not being supported or issues in the infrastructure (alas, not permissions most of the time)

The use of hard coded fake records within the calendar module used in the API tests makes that no longer possible, because those records are not available in the real module.

Which has meant a number of bugs have made it to the hosted environments that might have been identified earlier.

The rest of the test infrastructure has the records it needs set up during the test execution.

I also think it leads to more coupling and magic within the tests as special constants are used to refer to records for certain scenarios.

Does that make sense?

Reply:
Makes perfect sense. But since what I'm trying to develop here is a rather isolated piece of logic, which is not currently integrated into our API, is it even feasible to test it properly at this point?

To get started with the tests, I need a `Clients` object to build `CalendarRepository`, but I can't get it anywhere at this point, so I have to mock it, right? I there is another way to do this, please let me know.

This new service is just a puzzle piece that will become a part of circulation logic once [CIRC-524](https://issues.folio.org/browse/CIRC-524) is done. So maybe it would make more sense to test my changes (at least the part that calls the calendar) when they are integrated?

Reply:
> Makes perfect sense. But since what I'm trying to develop here is a rather isolated piece of logic, which is not currently integrated into our API, is it even feasible to test the calls to `mod-calendar` properly?
> 
> To get started with the tests, I need a `Clients` object to build `CalendarRepository`, but I can't get it anywhere at this point, so I have to mock it, right? I there is another way to do this, please let me know.

The service does not have to depend upon `Clients`. The intent behind clients was to provide a convenient place to find clients for different APIs. Originally, that followed the pattern that a factory method would use `Clients` however the class constructor would have repositories or individual clients as parameters.

Those can then be faked or mocked. Alternatively, specific interfaces can be introduced for the dependencies that your service actually needs.

Does that make sense?

> This new service is just a puzzle piece that will become a part of circulation logic once [CIRC-524](https://issues.folio.org/browse/CIRC-524) is done. So maybe it would make more sense to test my changes (at least the part that calls the calendar) within API tests when they are integrated?

There should be API tests for some of this once this is integrated and influences public observable behaviour.

There is an intent to move the emphasis of automated testing in mod-circulation to more unit like tests. With that in mind, I'd prefer that the code for this pull request was covered with more unit like tests where possible.

Does that make sense?

Reply:
> There should be API tests for some of this once this is integrated and influences public observable behaviour.

> With that in mind, I'd prefer that the code for this pull request was covered with more unit like tests where possible.

Just to make sure we are on the same page here: you are saying that I should cover everything except the calls to mod-calendar with unit tests? And cover the calendar interaction with API tests once this code is integrated?

Reply:
Yes, I think so. It would be great if there were unit like tests for this behaviour. If that means extracting classes and interfaces to fake / mock requests to storage then please do so.

When this is exposed as observable behaviour in the APIs then a few API tests that cover some (not all) scenarios can be added, to demonstrate the integration.

Hopefully that means that we can slow the rate of API test growth, and eventually start to shrink those tests in favour of more unit like tests.

Reply:
I rewrote the tests, please take a look.